pinot/net.py

```diff
@@ -109,24 +87,32 @@ def forward(self, g):
     def condition(self, g):
         """ Compute the output distribution.
         """
-        # get the parameters
-        theta = self.forward(g)
 
-        # put theta through noise model
-        theta = self.noise_model(theta)
+        if self.noise_model == 'normal-heteroschedastic':
+            mu, log_sigma = self.forward(g)
+            distribution = torch.distributions.normal.Normal(
+                    loc=mu,
+                    scale=torch.exp(log_sigma))
 
-        # parameterize the distribution according to the parameters
-        # and the distribution class
-        distribution = self.distribution_class(*theta)
+        elif self.noise_model == 'normal-homoschedastic':
+            mu, _ = self.forward(g)
 
-        return distribution
+            # initialize a `LOG_SIMGA` if there isn't one
+            if not hasattr(self, 'LOG_SIGMA'):
+                self.LOG_SIGMA = torch.tenosr(0.0)
+                self.LOG_SIMGA.requires_grad = True
 
-    def expectation(self, g):
-        # construct the distribution
-        distribution = self.condition(g)
-        
-        # expectation is the analytical mean of the distribution
-        return distribution.mean()
+            distribution = torch.distributions.normal.Normal(
+                    loc=mu,
+                    scale=torch.exp(self.LOG_SIGMA))
+
+        elif self.noise_model == 'normal-homoschedastic-fixed':
+            mu, _ = self.forward(g)
+            distribution = torch.distributions.normal.Normal(
+                    loc=mu, 
+                    scale=torch.tensor(1.0))
```

Comment:
similar to above, instead of 1.0 I'd use sth like ''' torch.ones(1,measurement_dim)"'' or whatever it is called

Reply:
also ```torch.ones(measurement_dim)``` will broadcast automagically in pytorch, it is just somewhat inconsistent with the other shapes